src/main/java/cart/data/ProductJdbcRepository.java

```diff
@@ -0,0 +1,85 @@
+package cart.data;
+
+import cart.data.entity.CartProduct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.simple.SimpleJdbcInsert;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class ProductJdbcRepository implements ProductRepository {
+
+    private final SimpleJdbcInsert simpleJdbcInsert;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    public ProductJdbcRepository(DataSource dataSource) {
+        this.simpleJdbcInsert = new SimpleJdbcInsert(dataSource).withTableName("CART_PRODUCT");
+    }
+
+    @Override
+    public void addProduct(CartProduct product) {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("product_name", product.getProductName());
+        parameters.put("product_price", product.getProductPrice());
+        parameters.put("product_image_url", product.getProductImageUrl());
+        parameters.put("created_at", product.getCreatedAt());
+        simpleJdbcInsert.execute(parameters);
+    }
+
+    @Override
+    public List<CartProduct> getProducts() {
+        JdbcTemplate jdbcTemplate = simpleJdbcInsert.getJdbcTemplate();
+        String sql = "select * from CART_PRODUCT";
+        return jdbcTemplate.query(sql, this.cartProductMapper());
+    }
+
+    @Override
+    public Optional<CartProduct> getProductById(long id) {
+        JdbcTemplate jdbcTemplate = simpleJdbcInsert.getJdbcTemplate();
+        String sql = "select * from CART_PRODUCT where product_id = ?";
+        CartProduct cartProduct = null;
+        try {
+            cartProduct = jdbcTemplate.queryForObject(sql, this.cartProductMapper(), id);
+        } catch (Exception e) {
+            logger.info("sql 오류 -> {}", e.getMessage());
+        }
+        return Optional.ofNullable(cartProduct);
```

Comment:
아래와 같이 반환해보는 것은 어떠신가요!
예외가 발생했을 때 비어있는 Optional 값을 전달하고
호출하는 쪽에서는 Optional 이 비어있는지 확인하고 사용하도록요 😄 

```suggestion
        try {
            return Optional.of(jdbcTemplate.queryForObject(sql, this.cartProductMapper(), id));
        } catch (Exception e) {
            return Optional.emtpy();
        }
```